### ISSUE TITLE
Error at boot time that eventually self-heal after a couple of reboots. Error resolving system account: account missing

### ISSUE THREAD

**Issue by org-ci-cd** [nats-io/nats-server#4185]
Hello guys ! 

I am facing an issue that happens from time to time when my Nats cluster node reboots (for CVE patching for example).
When server tries to boot, it fails with this error message:
**nats-server: error resolving system account: account missing**
After a couple of container restart, it finally succeed. 
It seems to maybe be an issue with resolving the "system_account" public key. 

I've been trying to trace that issue back to it's root. Here are some more details:
- I am not able to reproduce this issue in another environment (we have a big amount of users in the Nats cluster in that environment (in the /nsc folder) over 150k users inside ~15 accounts)
- Nats version 2.9.15. 
- The account resolving uses the IN MEMORY setting with the nats.conf instruction _include "mem-resolver.conf"_
- Cluster of 3 nodes, the issue happens on all 3 nodes.
- All 3 nodes mount an AWS EFS shared drive that contains the /nsc folder with all the users in there. The EFS does not seem to suffer though, throughput is around 20% pretty stable. The EFS is mounted using Kubernetes PV/PVC.
- The nats servers are kubernetes machines pretty much dedicated to host the nats container, minus some daemonset placement for collectors, instance type is c6g.xlarge (so they are ARM machines).
- When the issue occurs, the mem-resolver.conf seems to be present on the machine, and the system_account public key is well populated. 

Do you have an idea to lead my investigation further ? I'm thinking it might either be an internal bug in Nats, or something to do with the large /nsc folder because of the amount of users we have, but I am unsure where to look next. I'm thinking of trying to create a massive amount of users in a lab environment to try out this theory. I also have an upgrade to the most recent version in the pipeline, currently in earlier environments.

Thanks a lot for hints !

**Comment by derekcollison** [comment]
Could you share your actual server config?

**Comment by org-ci-cd** [comment]
Hello Derek ! Of course, here it goes 
 
```
#include "advertise/client_advertise.conf"
listen: 0.0.0.0:8001
pid_file: "/var/run/nats/nats.pid"
http: 8222
debug: ${var.nats_cluster_activate_debug}
trace: ${var.nats_cluster_activate_tracing}

cluster {

  name: ${var.cluster_name}
  port: 6222
  connect_retries: 30
  no_advertise: true

  # Authorization for route connections
  # Other server can connect if they supply the credentials listed here
  # This server will connect to discovered routes using this user
  authorization {
    user: ${var.authorization_route_user}
    password: ${var.authorization_route_pwd}
    timeout: 6 # Authorization timeout must be higher than SSL timeout
  }

  # TLS only for cluster routes
  tls {
    # Route cert
    cert_file: "/tls/nats-internal-tls/tls.crt"
    # Private key
    key_file:  "/tls/nats-internal-tls/tls.key"
    # Optional certificate authority verifying connected routes
    # Required when we have self-signed CA, etc.
    ca_file:   "/tls/nats-internal-tls/ca.crt"
    timeout: 4
  }

  # This server establishes routes with these server.
  # This server solicits new routes and Routes are actively solicited and connected to from this server.
  # Other servers can connect to us if they supply the correct credentials
  # in their routes definitions from above.
  # If servers do not exist but the routes are listed here 
  # (e.g. only 3 server are present in the cluster), then the existing servers will
  # log warnings of this form:
  # ' Error trying to connect to route: dial tcp localhost:6333: connect: connection refused'
  # this should not cause any functional issue.
  # If routes are not listed (with username/password) like below, then existing servers won't be able
  # to connect to unlisted servers after they've been added to the cluster during scaling.
  #
  # This list should contain exactly the maximum number of replicas for this cluster
  #
  # TL;DR : A bigger list here is OK. It will pollute logs. Accept it. Otherwise forget scaling.
  routes [
    %{for server_index in range(var.min_replicas)~}
    nats://${var.authorization_route_user}:${var.authorization_route_pwd}@nats-${server_index}.nats.${var.namespace}.svc.cluster.local:6222
    %{endfor~}
  ]

}

# Default is 2s, but if we encounter connection issues, increase it.
write_deadline: "5s"

ping_max: 1 # Close the connection quick if the client is not answering

# No need for CA file as we use a public authority
tls {
    cert_file: "/tls/nats-client-tls/tls.crt"
    key_file: "/tls/nats-client-tls/tls.key"
    timeout: 4
}

# This needs to be a relative path
include "mem-resolver.conf"
system_account: <system_account_public_key>


```

**Comment by derekcollison** [comment]
How about mem-resolver.conf?

**Comment by org-ci-cd** [comment]
Also prepared by the process that takes care of preparing the cluster

```
  ################################################################################
  # Generate definitions of accounts in the "mem-resolver.conf" file
  # and move the file in a the proper folder so nats-servers
  # can access it and reload their "in-memory" config.
  ################################################################################
  echo "> Regenerate and Sync mem-resolver.conf for nats-servers"
  nsc generate config --mem-resolver > /path/to/nfs/again/mem-resolver.conf
```

and used by the nats server at boot

```
#########################################################################
echo "Copy nats-server requirements from the shared drive before booting"
#########################################################################

# Nats-server needs mem-resolver or tenants won't be able to connect
cp -f /path/to/nfs/again/mem-resolver.conf /etc/nats-config/mem-resolver.conf
```

**Comment by org-ci-cd** [comment]
What gets me thinking is that those files are copied over from the NFS to the local container disk at boot time (by the nats container), and the Nats processes were not able to boot using them, but after 5 container restart it ended up booting properly. :thinking:  

And the operator was not regenerated in between, those files should be pretty much static. (I might have to double check that again by recrashing the servers lol) but looking at the logs what I say seems correct.

**Comment by derekcollison** [comment]
Can you share the actual file?

**Comment by org-ci-cd** [comment]
Can you propose a way to I can do it privately ?

**Comment by derekcollison** [comment]
Sure email if that works or DM on slack.

**Comment by derekcollison** [comment]
Any updates?
